aicsimageio/readers/lif_reader.py

```diff
@@ -381,31 +381,44 @@ def _get_coords_and_physical_px_sizes(
         scale_x, scale_y, scale_z, scale_t = image_short_info["scale"]
 
         # Handle Spatial Dimensions
+        # In general, we have learned that floating point math is hard....
+        # This block of code used to use `np.arange` with floats as parameters and
+        # it was causing errors. To solve, we generate the range with ints and then
+        # multiply by a float across the entire range to get the proper coords.
+        # See: https://github.com/AllenCellModeling/aicsimageio/issues/249
         if scale_z is not None:
-            coords[DimensionNames.SpatialZ] = np.arange(
-                0,
-                image_short_info["dims"].z * scale_z,
-                scale_z,
+            coords[DimensionNames.SpatialZ] = (
+                np.arange(
+                    0,
+                    image_short_info["dims"].z,
+                )
+                * scale_z
```

Comment:
Since this happens a bunch apparently, should we maybe make this a utility function so that we only have to comment about behavior in one place?

Reply:
I hate it when you're right.

Reply:
Will update